### ISSUE TITLE
Response status code does not indicate success: 404 (Not Found).

### ISSUE THREAD

**Issue by muhammad-azeem-confiz** [AzureADSamples/SinglePageApp-DotNet#6]
i am trying to use this sample with my application.

Getting Error in here..
app.UseWindowsAzureActiveDirectoryBearerAuthentication(
                new WindowsAzureActiveDirectoryBearerAuthenticationOptions
                {
                    Audience = ConfigurationManager.AppSettings["ida:Audience"],
                    Tenant = ConfigurationManager.AppSettings["ida:Tenant"]
                });
Error:
Response status code does not indicate success: 404 (Not Found).

here is my web.config
<add key="ida:Tenant" value="https://<tenantname>.onmicrosoft.com" />
<add key="ida:Audience" value="https://<tenantname>.onmicrosoft.com/<Application-Name>" />
<add key="ida:ClientID" value="<Client ID of <Application-Name>>" />

What i am Doing wrong ?

**Comment by asith-w** [comment]
i have same issue
Stack Trace: 
https://gist.github.com/asith-w/4c49a31514387d639997

**Comment by Doff3n** [comment]
You are both most likely not using the correct domain/tenant

In azure, go to Active Directory and Domains and use the full URL: [tenant]=[youname].onmicrosoft.com, use this as specified both in your config and app.js as tenant.

**Comment by dstrockis** [comment]
Doff3n is right. Please make sure you have spelled the tenant exactly right. The web api is attempting to fetch the signing keys from AAD and is unable to find them.

**Comment by DaneVinson** [comment]
I agree, most likely the tenant name isn't correct. I think the issue for most people is the confusing naming conventions used by Azure to convert account names into tenant names. I have Azure credentials that show {first}.{last}@{some_name}.com. The "ida:Tenant" I use for that AAD is {first}{last}{some_name}.onmicrosoft.com. I have access to a completely seperate AAD and there my credentials show {first}.{last}@{another_name}.onmicrosoft.com. The "ida:Tenant" in that scenario is set as {another_name}.onmicrosoft.com.

The value of you "ida:Tenant" setting is dependent on how your AAD is setup and there's no easy place I can find to get that setting.

**Comment by dstrockis** [comment]
I can see how this is confusing.  Probably the easiest way to reliably discover the name of your tenant is to create a user in that tenant via the Azure Mgmt Portal and inspect the domain of that user.

**Comment by DaneVinson** [comment]
That's a great idea. That would have saved me quite a bit of time a while back. Just after I posted the above I found a place to get the tenant value directly in the Azure portal. Azure Portal -> your AAD -> Domains. If you have custom domain(s) the tenant names will be one of those. If you are just using default settings you'll see the text "Your directory comes with a default domain, {your_name}.onmicrosoft.com...". The tenant value needed is "{your_name}.onmicrosoft.com".

**Comment by typlatt** [comment]
My issue was that my tenant in the web.config file was "https://mytenant.onmicrosoft.com" and needed to be "mytenant.onmicrosoft.com". Once I removed the HTTPS:// from the tenant value it worked perfect.

**Comment by Jacob-McKay** [comment]
Thank you for the discussion on this thread, I was getting the 404 mentioned, and figured I just missed something.  A few things have changed in the portal it appears, like I was asked for an application id url, and what is referred to in the readme shows up in the portal as a "client id".  I thought tenant name was the name of the app we created in the readme.  Ultimately I discovered it was <one_of_my_subscription_names>.onmicrosoft.com thanks to @dstrockis suggestion
